### PR TITLE
Refactor/app header

### PR DESCRIPTION
**Pull request by b12f**

### CODE REVIEW

**Comment by oliverguenther** [comment]
- [x] **Mobile menu is quite broken with buttons not working, and resizing also not behaving properly.**

Compare community (left) with your branch (right)

https://user-images.githubusercontent.com/459462/112323581-366b4880-8cb2-11eb-8bd3-87f39c108ae5.mp4

**Comment by oliverguenther** [comment]
- [x] **The style for the quick-add button got lost**

Screenshot from 2021-03-24 15-05-21

**Comment by oliverguenther** [comment]
- [x] **The search icon is moving when hovering**

Peek 2021-03-29 08-34

**Comment by b12f** [comment]
>     * The search icon is moving when hovering

I cannot reproduce this @oliverguenther are you sure you were on the latest commit?

**Comment by oliverguenther** [comment]
> > ```
> > * The search icon is moving when hovering
> > ```
> 
> I cannot reproduce this @oliverguenther are you sure you were on the latest commit?

This no longer happens to me as well. I probably wasn't on the latest commit then.

I'm on a836e5977e now and the mobile issues still exist for me: I cannot open any dropdown through the mobile-emulation of Chrome console which works fine on current dev

Also the placement of the `+` quick add menu is wrong now (unless I missed a changed requirement): 
Screenshot from 2021-03-30 16-09-58

**Comment by oliverguenther** [comment]
All my issues are now resolved, with the exception of the quick-add-menu position

File: lib/redmine/menu_manager/top_menu/projects_menu.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def render_projects_dropdown
   end
 
   def project_items
-    [project_index_item, project_new_item]
+    [project_index_item]
```
Comment (oliverguenther) [inline lib/redmine/menu_manager/top_menu/projects_menu.rb:56]: Did you discuss this with @wielinde whether we want to remove the New project item from the projects dropdown?

Comment (b12f) [reply lib/redmine/menu_manager/top_menu/projects_menu.rb:56]: Yep, that's why it is this way. It wouldn't be hard to put it back in (basically 2 or 3 reverse cherry picks), but we talked abou thaving the right side of the app-header for global items, and the left side for project-specific stuff

**Review by oliverguenther — APPROVED** [review]
LGTM. I addressed the teaser position in the linked PR that needs to be rebased + merged after this one.

The only open issue is the position of the quick-add-menu. Please double check that this is as required by Wieland and I'm fine with merging this

**Comment by oliverguenther** [comment]
One more thing I noticed that will probably also have broken on this PR. If you have a direct login provider set, this results in a link "Log in" to be rendered rather than a drop down.

To reproduce, set `OPENPROJECT_OMNIAUTH__DIRECT__LOGIN__PROVIDER=developer` and restart the dev server

**Comment by b12f** [comment]
> One more thing I noticed that will probably also have broken on this PR. If you have a direct login provider set, this results in a link "Log in" to be rendered rather than a drop down.
> 
> To reproduce, set `OPENPROJECT_OMNIAUTH__DIRECT__LOGIN__PROVIDER=developer` and restart the dev server

This link is fixed too https://github.com/opf/openproject/pull/9110/files#diff-1f23e87091776c2a9b8dae652e9c88a20708216f98a881f73557dd917a74efefR79

**Pull request merged**